src/aliceVision/matching/ArrayMatcher_hnswlib.hpp

```diff
@@ -178,6 +176,12 @@ class ArrayMatcher_hnswlib : public ArrayMatcher<Scalar, Metric>
     int dimension_{0};
     std::unique_ptr<SpaceInterface<DistanceType>> HNSWmetric;
     std::unique_ptr<HierarchicalNSW<DistanceType>> HNSWmatcher;
+
+    void reset() {
+        HNSWmetric.reset(nullptr);
+        HNSWmatcher.reset(nullptr);
+        dimension_ = 0;
+    }
```

Comment:
it's better to separate public/private methods from member:

```c++
private:
    void reset() 
    {
￼        HNSWmetric.reset(nullptr);
￼        HNSWmatcher.reset(nullptr);
￼        dimension_ = 0;
￼    }

private:
    int dimension_{0};
￼    std::unique_ptr<SpaceInterface<DistanceType>> HNSWmetric;		￼    
￼    std::unique_ptr<HierarchicalNSW<DistanceType>> HNSWmatcher;		￼    

```

Reply:
Just pushed this change!